pypiserver/_app.py

```diff
@@ -90,7 +88,8 @@ def favicon():
 
 @app.route("/")
 def root():
-    fp = request.custom_fullpath
+    parsed = urlparse(request.urlparts.scheme + "://" + request.urlparts.netloc)
+    fp = parsed.path.rstrip("/") + "/" + request.fullpath.lstrip("/")
```

Comment:
Nice! Totally agree with replacing the hook with a direct function call. Just a reminder, I think you intended to replace this with `fp = request_fullpath(request)` too? 

```suggestion
    fp = request_fullpath(request)
```